### PR TITLE
MQTTexample: ChibiOS support

### PR DESCRIPTION
**Pull request by danielinux**
minor modifications required to run a publisher based on the example code on ChibiOS

### CODE REVIEW

File: examples/mqttexample.c [hunk 1]
```diff
@@ -269,7 +269,8 @@ int err_sys(const char* msg)
     if (msg) {
         PRINTF("wolfMQTT error: %s", msg);
     }
-	exit(EXIT_FAILURE);
+	//exit(EXIT_FAILURE);
```
Comment (dgarske) [inline examples/mqttexample.c]: Hi Daniele,

Please revert this change and either override the exit() macro in mqttexamples.h or use the `NO_EXIT` define.

If you define `NO_EXIT` ii will accomplish the same thing in mqttexamples.h. You are welcome to add logic such this to the top of mqttexample.h or in wolfmqtt/mqtt_types.h:

```
#if !defined(NO_EXIT) && defined(WOLFMQTT_CHIBIOS)
    #define NO_EXIT
#endif
```

Thanks,
David Garske, wolfSSL

**Comment by danielinux** [comment]
Solved exit() as suggested. Indeed compiles fine on ChibiOS with -DNO_EXIT. Thanks for the review.
